### DISCUSSION TITLE
pnpm version

### DISCUSSION THREAD

**Discussion by RainyNight9** [umijs/umi#12312] [Q&A]
为什么要控制 pnpm version ？不应该控制好 node 版本就好吗？现在项目把 pnpm 锁定为某一版本，很不灵活啊

**Comment by fz6m** [comment]
你指的是 umi 仓库的 pnpm 版本吗，umi 源码用啥版本不影响用户用啥版本，另外 `packageManager` 这个设定现在不支持范围版本，并且官方也没有要放开支持范围版本的意思，同时 pnpm v9 已经强制检验版本一致性了，corepack 也是一样会按照这个版本来安装，所以包管理器版本在开发源码的时候必须一致是社区大势，取决不了个人。

**Reply by RainyNight9** [reply]
哎，又要搞个 corepack...

**Reply by RSS1102** [reply]
> 你指的是 umi 仓库的 pnpm 版本吗，umi 源码用啥版本不影响用户用啥版本，另外 `packageManager` 这个设定现在不支持范围版本，并且官方也没有要放开支持范围版本的意思，同时 pnpm v9 已经强制检验版本一致性了，corepack 也是一样会按照这个版本来安装，所以包管理器版本在开发源码的时候必须一致是社区大势，取决不了个人。

但是在 https://github.com/umijs/umi/blob/master/package.json#L124 里是否应该是
`  "engines": {
    "node": ">=14",
    "pnpm": ">=8.11.0"
  },`
  这里会限制用户使用固定版本的pnpm，而不是最低版本要求。
  参考和[pnpm](https://github.com/pnpm/pnpm/blob/main/package.json#L65)一致配置
